substrafl/strategies/fed_pca.py

```diff
@@ -0,0 +1,153 @@
+from substrafl.nodes.train_data_node import TrainDataNode
+from substrafl.remote import remote
+from substrafl.schemas import FedAvgAveragedState
+from substrafl.schemas import FedAvgSharedState
+from substrafl.schemas import StrategyName
+from substrafl.strategies import FedAvg
+
+logger = logging.getLogger(__name__)
+
+
+class FedPCA(FedAvg):
+    """Federated Principal Component Analysis strategy.
+    This strategy should be used only using TorchFedPCAAlgo
+
+    The goal of this strategy is to perform a principal component analysis (PCA) by
+    computing the eigen vectors with highest eigen values of the covariance matrices
+    of the data samples. We assume we have clients indexed by $j$, with $n_j$ data samples each.
+    We note :math:`N = \\sum_j n_j` the total number of samples. We denote :math:`D` the dimension of
+    the data, and :math:`K` the number of eigen vectors computed.
+
+    It is based on the Federated sub-space iteration algorithm described
+    here https://doi.org/10.1093/bioadv/vbac026 (algorithm 3 of the paper)
+
+    During the process, the local covariance matrices of each center are not shared.
+    However, the column-wise mean of each dataset is shared across the centers.
+
+    The Federated Subspace iteration is divded in three steps:
+
+    Step 1: for d= 1, ..., D, each center computes the mean of the $d$ component of
+    their dataset and share it to the central aggregator. The central aggregator averages
+    the local mean and send to all the clients the global column-wise means of data.
+
+    Step 2: Each center normalize their dataset by substracting the global mean column-wise
+    and compute the covariance matrix of their local data after mean-subtraction. We denote by $C_j$
+    the local covariance matrices.
+
+    We initialize eig_0: a matrix of size :math:`D \\times K` corresponding to the :math:`K` eigen
+    vectors we want to compute
+    Step 3: For a given number of rounds (rounds are labeled by $r$) we perform the following:
+        Step 3.1: each center :math:`j` computes  :math:`eig^r_j = C_j \\dot eig^{r-1}_j`
+        Step 3.2: the aggregator computes :math:`eig^r = \\frac{1}{N}\\sum_j n_j eig^r_j`
+        Step 3:3: the aggregator performs a QR decomposition: :math:`eig^r \\mapsto QR(eig^r)`
+                    and sahres :math:`eig^r` to all the clients
+
+    :math:`eig^r` will converge to the :math:`K` eigen-vectors of the global covariance matrix with
+    the high eign-values.
+    """
+
+    def __init__(self):
+        super(FedPCA, self).__init__()
+
+    @property
+    def name(self) -> StrategyName:
+        """The name of the strategy
+
+        Returns:
+            StrategyName: Name of the strategy
+        """
+        return StrategyName.FEDERATED_PCA
+
+    def predict(
+        self,
+        algo: Algo,
+        test_data_nodes: List[TestDataNode],
+        train_data_nodes: List[TrainDataNode],
+        round_idx: int,
+    ):
```

Comment:
```suggestion
    ) -> None:
```
default return type is Any